lib/sinon/stub-descriptor.js

```diff
@@ -0,0 +1,43 @@
+"use strict";
+
+var deprecated = require("./util/core/deprecated");
+var objectKeys = require("./util/core/object-keys");
+var spy = require("./spy");
+var wrapMethod = require("./util/core/wrap-method");
+
+// This is deprecated and will be removed in a future version of sinon.
+// We will only consider pull requests that fix serious bugs in the implementation
+function stubDescriptor(object, property, descriptor) {
+    var wrapper;
+
+    deprecated.printWarning(
+      "sinon.stub(obj, 'meth', fn) is deprecated and will be removed from" +
```

Comment:
Yeah, I totally agree.

Let's discuss that in a separate issue, I've got some ideas for the API and how to implement it, I'd love to help with this. Also, this PR LG(reat)TM.

Awesome work, sir 😄